src/sentry/ingest/ingest_consumer.py

```diff
@@ -87,43 +89,20 @@ def process_message(self, message):
             ip=remote_addr, data=data, project=project, sender=self.process_message
         )
 
+    def flush_batch(self, batch):
+        pass
+
+    def shutdown(self):
+        pass
```

Comment:
Maybe I am missing some context, but what is the advantage of using the batching consumer if you are not doing anything with the processed batch ?
process_message, in a batching consumer, is called once per message in the same way as you were calling the equivalent method in the SimpleConsumer. 
the added value of the batching consumer is that it would batch together processed messages and then send them all together to the flush_batch method that can do its job (generally save on storage) in a batched way.
Is your goal to take advantage of batched commit? Which in this case would be subject to process messages twice if the consumer dies before committing.

Reply:
Right now we are not in a position to efficiently batch the processing of messages (for the particular processing we do). We still get something out of moving to batching-kafka-consumer:

* Code closer to snuba
* Option naming aligned with snuba
* Commit on every nth message (was manually implemented before too)
* Commit after a max duration

Reply:
To expand a bit on what Markus wrote, a few reasons here:
1. BatchingKafkaConsumer already supports a certain set of features that we do not want to reinvent/reimplement: flexible commit management, partition assignment/revocation, metrics. We tried to do something similar with our SimpleConsumer, but then we saw that the implementation was basically converging to BatchingKafkaConsumer.
2. We _might_ want to do something meaningful in `flush_batch` soon: there's a chance that performance will not be good enough and one of the proposals was to do something smarter about sending signals in outcomes consumer, like doing signal handling in batches.
3. Processing messages twice is not such a big deal in our case, e.g. in ingest consumer we have a dedupe check anyway
4. In general, we decided not to invest resources in another alternative consumer implementation right now. There are talks about writing a new, better consumer library, and we're totally onboard with that. It's just that migrating from BatchingKafkaConsumer will probably be easier, since we'll also be migrating snuba.